src/workloads/sharding/ReshardCollection.yml

```diff
@@ -0,0 +1,154 @@
+    5. Running read and write operations on the collection after it has been resharded.
+
+GlobalDefaults:
+  Nop: &Nop {Nop: true}
+
+  Database: &Database test
+  # Collection0 is the default collection populated by the MonotonicLoader.
+  Collection: &Collection Collection0
+  Namespace: &Namespace test.Collection0
+
+  DocumentSize: &DocumentSize 10000  # = 10kB
+  DocumentSize80Pct: &DocumentSize80Pct 8000  # = 8kB
+  DocumentCount: &DocumentCount 100000  # for an approximate total of 1GB
+
+  ShardKeyValueMin: &ShardKeyValueMin 1
+  ShardKeyValueMax: &ShardKeyValueMax 100
+
+  ReadWriteOperations: &ReadWriteOperations
+  - OperationName: findOne
+    OperationCommand:
+      Filter: {_id: {^RandomInt: {min: 1, max: *DocumentCount}}}
+  - OperationName: updateOne
+    OperationCommand:
+      Filter: {_id: {^RandomInt: {min: 1, max: *DocumentCount}}}
+      Update: {$inc: {counter: 1}}
+      # Write operations can fail with a ShardInvalidatedForTargeting error response due to the
+      # routing information for the temporary resharding collection not being available on the
+      # donor shard. TODO: Investigate why mongos isn't automatically retrying on it.
+      ThrowOnFailure: false
+
+Clients:
+  # The reshardCollection command is expected to take a long to complete so we give the actor
+  # running it a much higher socket timeout.
+  ReshardCollection:
+    QueryOptions:
+      maxPoolSize: 1
+      socketTimeoutMS: 3600000  # = 1 hour
+
+Actors:
+- Name: CreateShardedCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationMetricsName: EnableSharding
+      OperationName: AdminCommand
+      OperationCommand:
+        enableSharding: *Database
+    - OperationMetricsName: ShardCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        shardCollection: *Namespace
+        # Hashed sharding will pre-split the chunk ranges and evenly distribute them across all of
+        # the shards.
+        key: {oldKey: hashed}
+        numInitialChunks: 60
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: LoadInitialData
+  Type: MonotonicLoader
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    BatchSize: 1000
+    Threads: 1
+    DocumentCount: *DocumentCount
+    Database: *Database
+    CollectionCount: 1
+    Document:
+      oldKey: {^RandomInt: {min: *ShardKeyValueMin, max: *ShardKeyValueMax}}
+      newKey: {^RandomInt: {min: *ShardKeyValueMin, max: *ShardKeyValueMax}}
+      counter: 0
+      padding: {^FastRandomString: {length: {^RandomInt: {min: *DocumentSize80Pct, max: *DocumentSize}}}}
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: ReshardCollection
+  Type: AdminCommand
+  Threads: 1
+  ClientName: ReshardCollection
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: admin
+    Operations:
+    # Until SERVER-53373 is implemented, donor shards aren't guaranteed to preserve enough history
```

Comment:
[note] Assuming this seems is acceptable to the reviewers, I'll put up a PR for https://github.com/visemet/dsi/commit/d3b93efcd3a2b0f8a3f62dc0f46adb44f0ab622c too.

Reply:
> If it could be done at the beginning of the test we could put in a js check on version before enabling or disabling.

We set the WTPreserveSnapshotHistoryIndefinitely failpoint at startup in our correctness testing.

https://github.com/mongodb/mongo/blob/f3b3164a196fa930b17ac2f703db433d8fc79f36/jstests/sharding/libs/resharding_test_fixture.js#L77-L79

We could do something similar for our performance testing too. The reason I had chosen not to is because we'd be pinning the oldest\_timestamp from before the 1GB of data was inserted. If that 1GB of data becomes 40GB of data then I think we're going to see durable history stressed in a way that wouldn't be representative of the actual behavior during resharding.

> Can we version this somehow? Once we disable this we won't be able to run this test against versions before SERVER-53373 goes in.

To save myself some time, I had a goal in my initial cut of this workload to not write a new actor in C++. I'm willing to give up successfully running the ReshardCollection.yml workload on earlier mongodb/mongo commits.